clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -12,552 +12,120 @@
  *   Broadcom, Inc. - initial API and implementation
  */
 
-import * as fs from "node:fs";
-import * as iconv from "iconv-lite";
-import * as Path from "node:path";
 import * as vscode from "vscode";
-import {
-  DOWNLOAD_QUEUE_LOCKED_ERROR_MSG,
-  INSTALL_ZOWE,
-  INVALID_CREDENTIALS_ERROR_MSG,
-  PATHS_USS,
-  PATHS_ZOWE,
-  PROFILE_NAME_PLACEHOLDER,
-  PROVIDE_PROFILE_MSG,
-  SETTINGS_CPY_SECTION,
-  UNLOCK_DOWNLOAD_QUEUE_MSG,
-  ZOWE_EXT_MISSING_MSG,
-} from "../../constants";
-import { TelemetryService } from "../reporter/TelemetryService";
-import { SettingsService } from "../Settings";
+import { DownloadStrategyResolver } from "./downloader/DownloadStrategyResolver";
+import { CopybookDownloader } from "./downloader/CopybookDownloader";
+import { PROVIDE_PROFILE_MSG } from "../../constants";
 import { ProfileUtils } from "../util/ProfileUtils";
-import { Utils } from "../util/Utils";
-import { CopybookURI } from "./CopybookURI";
-import { CopybookProfile, DownloadQueue } from "./DownloadQueue";
-import { TelemetryEventMeasurements } from "../reporter/model/TelemetryEvent";
+import { ZoweExplorerDownloader } from "./downloader/ZoweExplorerDownloader";
 
 export class CopybookName {
-  public constructor(public name: string, public dialect: string) {}
+  constructor(public name: string, public dialect: string) {}
 }
 
-const experimentTag = "experiment-tag";
-export class CopybookDownloadService implements vscode.Disposable {
-  public constructor(public storagePath: string) {}
-  public static checkWorkspace(): boolean {
-    const workspaceFolders = vscode.workspace.workspaceFolders!;
-    if (workspaceFolders && workspaceFolders.length === 0) {
-      vscode.window.showErrorMessage("No workspace folder opened.");
-      return false;
-    }
-    return true;
-  }
-
-  private static completedDownload: number = 0;
-  private static totalDownload: number = 0;
-
-  private static processDownloadError(title: string) {
-    const actionSettings = "Change settings";
-    vscode.window.showErrorMessage(title, actionSettings).then((action) => {
-      if (action === actionSettings) {
-        vscode.commands.executeCommand("cobol-lsp.cpy-manager.goto-settings");
-      }
-    });
-  }
-
-  private static needsUserNotification(queue: CopybookProfile[]): boolean {
-    return queue.find((profile) => !profile.quiet) !== undefined;
-  }
+export class CopybookDownloadService {
+  private downloadResolver: CopybookDownloader;
```

Comment:
```suggestion
  private downloadResolver: DownloadStrategyResolver;
```
+ cleanup `downloadResolver!`

Reply:
updated